### PR TITLE
SECD-621 Implement DynamoDB scan timestamp storer and fetcher

### PR DESCRIPTION
**Pull request by bkrebsbach**
* Add an implementation for the TimestampFetcher/TimestampStorer interfaces which uses a DynamoDB table as the underlying storage mechanism
  * Persist and retrieve scans from a DynamoDB table with a static partition key
  * Add a settings component to configure DynamoDB connection
  * Does NOT add any integration tests
* Add implementation to handler in main.go
* Load correct file in gateway-outbound.Dockerfile
* Add new environment variables to docker-compose.yaml

### CODE REVIEW

File: docker-compose.yaml [hunk 1]
```diff
@@ -5,28 +5,36 @@ services:
       context: .
       dockerfile: Dockerfile
     environment:
-      - SERVERFULL_RUNTIME_HTTPSERVER_ADDRESS=:8081
```
Comment (LorenBoren) [inline docker-compose.yaml:8]: Question: I think a lot of our docker-compose.yml files in our repos have this format, if it's wrong, I think the other repos should be fixed as well. I couldn't find docs showing that one particular way was right or wrong though, can you share the doc if you found it?

Comment (bkrebsbach) [reply docker-compose.yaml:8]: It's not wrong, both formats are acceptable in the spec. I find this format simpler to troubleshoot when there are issues with the environment variables using YAML linters and syntax highlighting. The other list format treats both the key and value as a single string, so it's a little harder for me to detect problems. If you've got a strong preference, I'll revert it back.

**Review by faldridge — DISMISSED** [review]

File: gateway-outbound.Dockerfile [hunk 2]
```diff
@@ -1,3 +1,3 @@
 FROM asecurityteam/serverfull-gateway
-COPY api.yaml .
+COPY api-outbound.yaml .
```
Comment (kconwayatlassian) [inline gateway-outbound.Dockerfile:2]: +1 on the name change. inbound/outbound seems much more specific than external/internal

**Review by kconwayatlassian — CHANGES_REQUESTED** [review]
There's seems to be an implied schema for the table. Would you add that schema to the README since folks will need to create it out-of-band of the system?

File: gateway-outbound.Dockerfile (hunk 2 above)
Comment (bkrebsbach) [reply gateway-outbound.Dockerfile:2]: Should I change api.yaml to api-inbound.yaml?

Comment (kconwayatlassian) [reply gateway-outbound.Dockerfile:2]: I, personally, find it clearer and would appreciate it.

Comment (bkrebsbach) [reply gateway-outbound.Dockerfile:2]: Fixed

**Comment by bkrebsbach** [comment]
@kconwayatlassian I can't reply to your original message, but I've added the schema to the README

**Review by kconwayatlassian — APPROVED** [review]

**Review by zlozano — APPROVED** [review]

File: docker-compose.yaml (hunk 1 above)
Comment (LorenBoren) [reply docker-compose.yaml:8]: I don't have a preference for the formatting, but I would prefer if things like this be consistent across our repos. I don't want this to be a blocker for merging. Maybe we could have a separate conversation about this.

Comment (bkrebsbach) [reply docker-compose.yaml:8]: Sure, I'd be happy to.

**Review by LorenBoren — APPROVED** [review]

**Pull request merged**